packages/flutter_tools/lib/src/commands/update_packages.dart

```diff
@@ -644,23 +648,23 @@ class PubspecYaml {
         //
         // Try to parse the line by giving it to the last PubspecDependency
         // object we created. If parseLock fails to recognize the line, it will
-        // throw. If it does recognize the line but decides it's one we don't
-        // care about (specifically, "git:" dependencies), it'll return false.
+        // throw. If it does recognize the line and needs the following lines in
+        // its lockLine, it'll return false.
         // Otherwise it returns true.
         //
         // If it returns true, then it will have updated itself internally to
         // store the information from this line.
         if (!lastDependency.parseLock(line, filename, lockIsOverride: section == Section.dependencyOverrides)) {
-          // Ok we're dealing with some "git:" dependency. Let's pretend we
-          // never saw it. In practice this is only used for the flutter gallery
-          // assets dependency which we don't care about especially since it has
-          // no subdependencies and it's pinned by git hash.
-          //
-          // Remove the PubspecDependency entry we had for it and replace it
-          // with a PubspecLine entry, and add such an entry for this line.
-          result.removeLast();
-          result.add(PubspecLine(lastDependency.line));
-          result.add(PubspecLine(line));
+          // Ok we're dealing with some "git:" dependency. Consume lines until
+          // we are out of the git dependency, and stuff them into the lock
+          // line.
+          lastDependency._lockLine = line;
+          lastDependency._lockIsOverride = section == Section.dependencyOverrides;
+          do {
+            index += 1;
+            line = lines[index];
+            lastDependency._lockLine += '\n$line';
```

Comment:
I could make a method on `PubspecDependency` to handle this, but  .. this whole thing is pretty convoluted as it is, and all the classes are in this file anyway.